include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+            forcing_vector_index = 0;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Forcing vector index is less than zero. Therefore, setting index to zero." << std::endl;
+        }
+
+        //Check if forcing index is greater than or equal to the size of the size of the time vector and if so, set to zero.
+        else if (forcing_vector_index >= time_epoch_vector.size())
+        {
+            forcing_vector_index = time_epoch_vector.size() - 1;
+            /// \todo: Return appropriate warning
+            std::cout << "WARNING: Reached beyond the size of the forcing vector. Therefore, setting index to last value of the vector." << std::endl;
+        }
+
+        return;
+    }
+
+    /**
+     * Get the current value of a forcing param identified by its name.
+     *
+     * @param name The name of the forcing param for which the current value is desired.
+     * @param index The index of the desired forcing time step from which to obtain the value.
+     * @return The particular param's value at the given forcing time step.
+     */
+    inline double get_value_for_param_name(const std::string& name, int index) {
+        if (index >= time_epoch_vector.size() ) {
+            throw std::out_of_range("Forcing had bad index " + std::to_string(index) + " for value lookup of " + name);
+        }
+
+        std::string can_name = name;
+        if(data_access::WellKnownFields.count(can_name) > 0){
+            auto t = data_access::WellKnownFields.find(can_name)->second;
+            can_name = std::get<0>(t);
+        }
+
+        if (forcing_vectors.count(can_name) > 0) {
+            return forcing_vectors[can_name].at(index);
+        }
+        else {
+            throw std::runtime_error("Cannot get forcing value for unrecognized parameter name '" + name + "'.");
+        }
+    }
+
+    /**
+     * @brief Read Forcing Data from CSV
+     * Reads only data within the specified model start and end date-times.
+     * @param file_name Forcing file name
+     */
+    // No file to read
+    void read_csv()
+    {
+                //std::string var_name = CSDMS_STD_NAME_RAIN_VOLUME_FLUX;
+                available_forcings.push_back("time");
+                std::string var_name = "precip";
+                std::string units = "";
+                auto wkf = data_access::WellKnownFields.find(var_name);
+                if(wkf != data_access::WellKnownFields.end()){
+                    units = units.empty() ? std::get<1>(wkf->second) : units;
+                    available_forcings.push_back(var_name); // Allow lookup by non-canonical name
+                    available_forcings_units[var_name] = units; // Allow lookup of units by non-canonical name
+                    var_name = std::get<0>(wkf->second); // Use the CSDMS name from here on
+                }
+    }
+
+    std::vector<std::string> available_forcings;
+    std::unordered_map<std::string, std::string> available_forcings_units;
```

Comment:
If `read_csv` is removed then this variable will be unused and should be removed.

Reply:
We cannot remove read_csv as it stands now. May be a better name is needed. It seems some non-empty string variable is needed in the initialization process.